File: app/src/main/java/com/klinker/android/twitter_l/data/roomdb/daos/DraftDao.java
package com.klinker.android.twitter_l.data.roomdb.daos;


import java.util.List;


import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.klinker.android.twitter_l.data.roomdb.entities.Draft;

@Dao
public interface DraftDao {

    @Insert
    void insertDraft(Draft draft);

    @Delete
    void deleteDraft(Draft draft);

    @Query("DELETE FROM drafts WHERE text = :message")
    void deleteDraft(String message);


    @Query("SELECT * FROM drafts WHERE account = :account")
    List<Draft> getDrafts(int account);


}

File: app/src/main/java/com/klinker/android/twitter_l/data/roomdb/daos/DraftDao.kt
package com.klinker.android.twitter_l.data.roomdb.daos


import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

import com.klinker.android.twitter_l.data.roomdb.entities.Draft

@Dao
abstract class DraftDao {

    fun createDraft(text: String, account: Int) : Draft {
        return Draft(text = text, account = account)
    }

    fun saveDraft(text: String, account: Int) : Draft? {
        return saveDraft(createDraft(text, account))
    }

    fun saveDraft(draft: Draft) : Draft? {
        return insertDraft(draft)?.let { id -> draft.copy(id = id) }
    }

    @Insert
    abstract fun insertDraft(draft: Draft): Long?

    @Delete
    abstract fun deleteDraft(draft: Draft)


    @Query("SELECT * FROM drafts WHERE account = :account ORDER BY id DESC LIMIT :pageSize OFFSET ((:page - 1) * :pageSize)")
    abstract fun getDrafts(account: Int, page: Int = 1, pageSize: Int = 50): List<Draft>


    @Query("DELETE FROM drafts WHERE account = :account")
    abstract fun deleteDrafts(account: Int)

}
